manifester/helpers.py

```diff
@@ -120,6 +124,34 @@ def fetch_paginated_data(manifester, endpoint):
         return _endpoint_data
 
 
+def load_inventory_file(file):
+    """Load local inventory file.
+
+    :return: list of dictionaries
+    """
+    if file.suffix in (".yaml", ".yml"):
+        loader = yaml
+        loader_args = {"Loader": yaml.FullLoader}
+        with file.open() as f:
+            return loader.load(f, **loader_args) or []
+    else:
+        logger.warn(
+            f"Found invalid inventory file {file}. Inventory file must exist and "
+            "have a .yaml or .yml suffix."
+        )
+
+
+def update_inventory(inventory_data):
+    """Replace the existing inventory file with current subscription allocations."""
+    inventory_path = Path(settings.inventory_path)
+    if load_inventory_file(inventory_path):
```

Comment:
this check also loads the inventory but does nothing with the data. if you don't want to keep the data, then it's easier to just check if the file exists.
however, since this function is called update_inventory, and you're checking an inventory_data variable below, you likely want to store the output.

Reply:
Thanks, this is a good point. 

I've been trying to think of a case in which I would want `update_inventory` to keep the existing inventory data rather than just clobbering it with the latest from the API, but I have come up short so far. So I'm considering renaming the function to `replace_inventory` and switching to just checking if the inventory file exists rather than loading it. I would keep the `load_inventory` function in place for use with the CLI, which I'm planning to work on after this PR. 

What do you think about this idea?

Reply:
in most cases, for manifester, a destructive change likely isn't an issue, but in the case that something goes wrong with the sync process, then a user will be left with a blank inventory (valid or not). it wouldn't be a bad idea to old onto the current inventory data as a fallback.

Reply:
@JacobCallahan I'm not sure if something like this is what you had in mind, so I'm interested in your thoughts on this change. As currently written, getting a manifest causes the inventory to get updated several times per run, so updating the backup file on each pass should ensure that the most recent good data is not lost in the event of an issue. 

I also think it's unlikely that an error will make it through the exception handling in the `simple_retry` function, although it's tricky to test this with the live API. From my perspective, if there is an error that we're seeing make it through to `update_inventory`, it would be preferable to expand `simple_retry`'s exception handling rather than relying on stored inventory data. What do you think?

Reply:
I'm personally not a fan of making a backup file over just loading the current contents into memory and dumping back into the file if something goes wrong.

Reply:
@JacobCallahan That's fair enough; I dropped that commit. 

However, I still think, since `update_inventory` is only responsible for modifying the local inventory file, it would be preferable to preserve the existing inventory by preventing `update_inventory` from running if there is an issue with getting the inventory data. Anything that can go wrong with requesting the data will have gone wrong before `update_inventory` executes. `simple_retry` should catch HTTP 429, 500, and 504 responses, so my latest commit adds handling of some other error responses to `fetch_paginated_data`.

I don't see a way to stash the inventory data in a useful way without making substantial changes to the control flow of the inventory update. That's something that I could theoretically undertake, but it seems out of scope for the current effort. Let me know what you think.

Reply:
ACK on not expanding scope at this time.